### DISCUSSION TITLE
Errors while running jenkins jobs

### DISCUSSION THREAD

**Discussion by NithyaParavasthu** [Sunbird-Ed/Community#111] [Bugs]
**Good afternoon.**

### I am facing issues when running Jenkins jobs,
- Some jobs are throwing errors saying **JDK issue** even when we installed java in the node. 
- Some jobs are throwing **groovy** errors.
- Some jobs throw errors regarding **MVN** not found, and unrecognized repository tags. Maven is installed in the node
- Some jobs are not running because some of the **default plugins** cannot install without restart, and when restarted all the jobs are getting washed out.  
- Some jobs are asking for **parameters**
- Some jobs showing **Jenkins files** not found.

I used the following repositories and documentation for reference. 
- http://docs.sunbird.org/4.6.0/developer-docs/
- https://github.com/Sunbird-Ed/SunbirdEd-portal
- https://github.com/project-sunbird/sunbird-devops

### I am attaching some screen shorts for your reference 
![image](https://user-images.githubusercontent.com/91603971/187165475-6e26874a-28d7-4f2c-b19b-08c1c25c58a7.png)
![image](https://user-images.githubusercontent.com/91603971/187165950-00db3f9e-e759-4f28-92ac-a0c6a68e6668.png)
![image](https://user-images.githubusercontent.com/91603971/187166079-06e5587d-5eb8-4d74-8d3b-70e71199d079.png)
![image](https://user-images.githubusercontent.com/91603971/187166189-98fa0653-fe61-42a0-8332-3f7b8b57ac4d.png)
![image](https://user-images.githubusercontent.com/91603971/187166334-cb7f0279-63f6-4c20-ad32-1a2eeb8c40f4.png)

**Comment by Ganeshpottola** [comment]
Even I'm facing the same issue, Can anyone help me with this issue

**Comment by beepdot** [comment]
Have you done all the steps mentioned here? http://docs.sunbird.org/4.6.0/developer-docs/server-installation/jenkins-setup/ ?

**Reply by NithyaParavasthu** [reply]
yes. I followed all the steps mentioned in the document and when I was running the Jenkins jobs I was facing the issues I mentioned above

**Reply by beepdot** [reply]
Do you have a environment variable named JAVA11_HOME on jenkins? What is the value set of it?
Do you have Java 11 installed on jenkins. Can you please share the output of below command from jenkins VM
```
/usr/lib/jvm/java-11-openjdk-amd64/bin/java -version
```

When running jobs (example `ApplicationElasticsearch`), are you providing a value for `branch_or_tag` parameter in Jenkins or leaving it empty?

**Comment by sharathkashyap** [comment]
@NithyaParavasthu Did input from @keshavprasadms help you

**Comment by sharathkashyap** [comment]
@NithyaParavasthu May have to lock this thread due to inactivity. Kindly open the new thread if you see that issue still exists. 
cc: @keshavprasadms @Ganeshpottola
